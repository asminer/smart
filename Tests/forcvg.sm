
/*
    $Id$
 */
	
/*
    Tests a for - converge nesting

*/

// # Verbose true
# ConvergePrecision 1e-7
	
for (int i in {1..10}) {
  converge {
    real x[i] guess 100*i; 
    real x[i] := sqrt(i*x[i]);
  }
  print("x[", i, "] converged to ", x[i], "\n"); 
}

print("Multiple guess error, for arrays\n");
for (int i in {8..10}) {
  converge {
    real y[i] guess i*10;
    real y[i] guess i*10+5;
    real y[i] := sqrt(y[i]);
    real w[i] := sqrt(y[i]);
    real w[i] guess i;
  }
  print("y[", i, "] converged to ", y[i], "\n");
  print("w[", i, "] converged to ", w[i], "\n");
}

print("Multiple definition error, for arrays\n");
for (int i in {8..10}) {
  converge {
    real z[i] guess i*10;
    real z[i] := sqrt(z[i]);
    real z[i] := sqrt(z[i]+1);
  }
  print("z[", i, "] converged to ", z[i], "\n");
}

print("Wide split converge error for arrays\n");

for (int i in {8..10}) {
  converge {
    real widesplit[i] guess i*10;
  }
}
for (int i in {8..10}) {
  converge {
    real widesplit[i] := sqrt(widesplit[i]);
  }
}
for (int i in {8..10}) {
  print("widesplit[", i, "] should be ", i*10, ": ", widesplit[i], "\n");
}

print("Near split converge error for arrays\n");

for (int i in {8..10}) {
  converge {
    real nearsplit[i] guess i*10;
  }
  converge {
    real nearsplit[i] := sqrt(nearsplit[i]);
  }
}
for (int i in {8..10}) {
  print("nearsplit[", i, "] should be ", i*10, ": ", nearsplit[i], "\n");
}

