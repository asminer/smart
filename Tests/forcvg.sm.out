x[1] converged to 1
x[2] converged to 2
x[3] converged to 3
x[4] converged to 4
x[5] converged to 5
x[6] converged to 6
x[7] converged to 7
x[8] converged to 8
x[9] converged to 9
x[10] converged to 10
Multiple guess error, for arrays
ERROR in file forcvg.sm near line 26:
	Duplicate guess for identifier y
ERROR in file forcvg.sm near line 29:
	Array w already defined
y[8] converged to 1
w[8] converged to 1
y[9] converged to 1
w[9] converged to 1
y[10] converged to 1
w[10] converged to 1
Multiple definition error, for arrays
ERROR in file forcvg.sm near line 40:
	Array z already defined
z[8] converged to 1
z[9] converged to 1
z[10] converged to 1
Wide split converge error for arrays
ERROR in file forcvg.sm near line 54:
	Array widesplit already defined
widesplit[8] should be 80: 80
widesplit[9] should be 90: 90
widesplit[10] should be 100: 100
Near split converge error for arrays
ERROR in file forcvg.sm near line 68:
	Array nearsplit already defined
nearsplit[8] should be 80: 80
nearsplit[9] should be 90: 90
nearsplit[10] should be 100: 100
